ci/download-rust.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+tmp=$(mktemp -d)
+export GNUPGHOME="$tmp"
+gpg --import <<EOF
```

Comment:
should we maybe leave a comment about this public key? what is it, where to get it?

Reply:
That is a good point. I have expanded the comments in general including your points and done some refactoring inside cc9b0be90944abc4bd3aad3473be2dc6ac18b683.